### DISCUSSION TITLE
Unable to install MetaFieldsBundle-1.18 plugin

### DISCUSSION THREAD

**Discussion by blessleness** [kimai/kimai#3022] [On-Premise: Installation & Updates] [answered]
Dear community

This is my first post on Github, I'm not that familiar with it...

I discovered Kimai today which seems to be quite in adequation with my needs to follow the activity of my team (with some tweaking).

I installed it on a VM running Ubuntu 21.10, followed the instructions provided online, I managed to get Kimai to work. However, it occus that having the MetaFieldsBundle plugin would allow me to go even further, so I purchased it, and now I'm stuck... I think I followed instructions properly, copying folder into /var/plugins/MetaFieldsBundle, then applying chown -R :www-data on the site, chmod rules etc.
Then I cleaned the cache and excecuted the command to add stuff in the DB...

```
blessleness@ubuntu:/var/www/kimai2/var/plugins/MetaFieldsBundle$ ll
total 168
drwxrwxr-x 15 root www-data  4096 Dec 14 08:22 ./
drwxrwxr-x  3 root www-data  4096 Dec 14 08:25 ../
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 API/
-rw-rw-r--  1 root www-data  3954 Dec 14 08:22 CHANGELOG.md
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Command/
-rw-rw-r--  1 root www-data  1845 Dec 14 08:22 composer.json
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Controller/
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 DependencyInjection/
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Entity/
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 EventSubscriber/
drwxrwxr-x  3 root www-data  4096 Dec 14 08:22 Form/
drwxrwxr-x  3 root www-data  4096 Dec 14 08:22 .github/
-rw-rw-r--  1 root www-data    52 Dec 14 08:22 .gitignore
-rw-rw-r--  1 root www-data  3888 Dec 14 08:22 LICENSE
-rw-rw-r--  1 root www-data 60205 Dec 14 08:22 meta-fields-bundle-logo.png
-rw-rw-r--  1 root www-data   453 Dec 14 08:22 MetaFieldsBundle.php
-rw-rw-r--  1 root www-data  4602 Dec 14 08:22 MetaFieldsRegistry.php
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Migrations/
-rw-rw-r--  1 root www-data  6127 Dec 14 08:22 .php-cs-fixer.dist.php
-rw-rw-r--  1 root www-data   523 Dec 14 08:22 phpstan.neon
-rw-rw-r--  1 root www-data  7062 Dec 14 08:22 README.md
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Repository/
drwxrwxr-x  5 root www-data  4096 Dec 14 08:22 Resources/
drwxrwxr-x  2 root www-data  4096 Dec 14 08:22 Twig/
drwxrwxr-x  3 root www-data  4096 Dec 14 08:22 Validator/
```

Each time I try to install it, my entire Kimai site is down telling me that "an error occured..." with a button to the dashboard that does not do anything... I cannot reach the dashboard any longer.

In the prod.log I see some error (quite a bunch actually) like 

-  Failed to save key "getTypes.a%3A3%...
- prod/pools/65Ui8J-V3o/c4a7fd7a19ae): Failed to open stream: Permission denied at /var/www/kimai2/vendor/symfony/cache/Traits/FilesystemCommonTrait.php:99)"}
- [2021-12-14 09:25:18] security.DEBUG: Guard authenticator does not support the request. {"firewall_key":"secured_area","authenticator":"App\\Security\\TokenAuthenticator"} []

Well I don't know whether this may help you... anyway, should you have some piece of advise, it would be highly appreciated...

Thanks in advance

**Comment by kevinpapst** [comment, marked as answer]
Hey, thanks for supporting Kimai!

> Failed to open stream: Permission denied

This tells you exactly what is wrong: the directory permissions - Kimai cannot create cache files.
See the second part of https://www.kimai.org/store/custom-fields-bundle.html#installation - you need to set the directory permissions after each cache clear!

You might want to scroll down to the bottom of this page: https://www.kimai.org/documentation/cache.html
There is a script that helps you to reload the cache and fix the file permissions afterwards in one step.

**Comment by blessleness** [comment]
BIG THANKS Kevin

I actually used the script provided, it solved all the issues !

Now everything seems to be working fine!

Thanks a lot
